File: cpp/Automation.PageObject.Example/Application/Google/Pages/SearchResults/WebSearchResults.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Automation.Common.Initialization.Interfaces;

namespace Automation.PageObject.Example.Application.Google.Pages.SearchResults
{
    public class WebSearchResults : GoogleBasePage
    {
        public WebSearchResults(ITestConfiguration testConfig) : base(testConfig)
        {
            RegisterSubElement("Submit Query", new { name = "btnG" });
            RegisterSubElement("Selected Web Tab", new { TagName = "div", Text = "Web", @class = "contains=hdtb-msel" });

            EnsureElementLoaded("Selected Web Tab", "Google web search page loaded.", "Google web search page failed to load.");
        }
    }
}

File: js/example/src/application/Google/Pages/SearchResults/WebSearchResults.js
var __extends = (this && this.__extends) || function (d, b) {
    for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p];
    function __() { this.constructor = d; }
    d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
};
var Google = require("../GoogleBasePage");
var WebSearchResults = (function (_super) {
    __extends(WebSearchResults, _super);
    function WebSearchResults(testConfig) {
        _super.call(this, testConfig);
        this.RegisterSubElement("Submit Query", { name: "btnG" });
        this.RegisterSubElement("Selected Web Tab", { TagName: "div", Text: "Web", "class": "contains=hdtb-msel" });
        this.EnsureElementLoaded("Selected Web Tab", "Google web search page loaded.", "Google web search page failed to load.");
    }
    return WebSearchResults;
})(Google.GoogleBasePage);
exports.WebSearchResults = WebSearchResults;
